pkg/controllers/termination/terminate.go

```diff
@@ -135,7 +131,7 @@ func (t *Terminator) evict(ctx context.Context, pods []*v1.Pod) {
 	critical := []*v1.Pod{}
 	nonCritical := []*v1.Pod{}
 	for _, pod := range pods {
-		if !pod.DeletionTimestamp.IsZero() {
+		if pod.DeletionTimestamp != nil {
```

Comment:
Any reason to not use the IsZero() function here?

Reply:
Yep! It's actually the wrong semantic. If the DeletionTimestamp is truly 0 (e.g. 1969 or w/e), then we want to behave differently. Zero and Nil are different for our use case.

Reply:
So weird. Thanks!